test/modules/obsagent-client/src/otel_client.rs

```diff
@@ -0,0 +1,130 @@
+use std::{
+    error::Error,
+    sync::{
+        atomic::{AtomicBool, Ordering},
+        Arc, Mutex,
+    },
+    time::Duration,
+};
+
+use futures::{Stream, StreamExt};
+use opentelemetry::{
+    global,
+    metrics::{self, ObserverResult},
+    sdk::metrics::{selectors, PushController},
+};
+use opentelemetry_otlp::ExporterConfig;
+use rand::Rng;
+use tracing::error;
+
+use crate::config::Config;
+
+const METER_NAME: &str = "microsoft.com/azureiot-edge";
+
+// Skip first immediate tick from tokio, not needed for async_std.
+fn delayed_interval(duration: Duration) -> impl Stream<Item = tokio::time::Instant> {
+    opentelemetry::util::tokio_interval_stream(duration).skip(1)
+}
+
+fn init_meter(period: f64, otlp_endpoint: String) -> metrics::Result<PushController> {
+    let export_config = ExporterConfig {
+        endpoint: otlp_endpoint,
+        ..ExporterConfig::default()
+    };
+    opentelemetry_otlp::new_metrics_pipeline(tokio::spawn, delayed_interval)
+        .with_export_config(export_config)
+        .with_aggregator_selector(selectors::simple::Selector::Histogram(vec![
+            0.1, 0.2, 0.3, 0.4, 0.5, 0.6, 0.7, 0.8, 0.9,
```

Comment:
I suppose 9 bins are fine. I'm not sure how cluster the data might be. It maybe a good idea if it's customable? (This might be just be nit-picking).

Reply:
Discussed offline with @yophilav and agreed to consider this change if/when a need arises in the future. For now the hard-coded bins are sufficient for the intended use case.